### PR TITLE
Test Delete and Add functions

### PR DESCRIPTION
**Pull request by benwmx**
### First Capstone
the fulfilled requirements : 
* Add a test to Add function
* Add a test to Delete function

#### Optional requirements : 
* Add ```--coverage``` flag to  jest to cover the exact percentage of tested files.

test-pic

### CODE REVIEW

File: src/test/task.test.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  tasks = removeTask(tasks, 2);
+  // Assert
+  test('Delete an item', () => {
+    expect(tasks.length).toBe(2);
+  });
+  test('Delete the exact item', () => {
+    expect(tasks[0]).toMatchObject(output[0]);
+    expect(tasks[1]).toMatchObject(output[1]);
+  });
+});
```
Comment (Emmyn5600) [inline src/test/task.test.js:41]: - [ ] Great Implementation, all tests are passing successfully, however, let us add few tests for remove function
arr14

`For Deleting or remove function`

arr15

Update index after task is deleted
Update localStorage after deleting  elements
Update dom after deleting compeleted elements

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUESTED :recycle:

Hi, @benwmx and @ahzia 

Great work on implementing the project requirements, however there are some issues to be solved before you get approved but you are almost there:muscle:. 
   
- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- All tests are passing successfully:heavy_check_mark:

Changes required :writing_hand: 


      * #### Please check the comments under the review.
       
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/test/task.test.js (hunk 1 above)
Comment (ahzia) [reply src/test/task.test.js:41]: @Emmyn5600 Thanks for the Review.
According to Requirements:

> Take a look at each of these functions. **Are they pure functions? If the answer is "yes"** then writing tests for them should be straightforward. 

Our functions are Pure Functions, We have different Functions for Updating local storage, and manipulating the dom, and they are separated from Remove Functions; So writing test for Remove is straight forward and we don't need to write the test for local storage and dom manipulations as they are different units (Functions).

Comment (Emmyn5600) [reply src/test/task.test.js:41]: Hi @ahzia

`sample for update localstorage after deleting elements`

  test('Update localStorage after deleting  elements', () => {
    clearCompletedListener(tasksList, container)(eventMock);
    const fetchedList = localStorage.getItem(LIST_STORAGE_KEY);
    expect(fetchedList).toEqual(JSON.stringify(tasksList.list));
  });

**Comment by Temmarie** [comment]
Hi @benwmx and @ahzia 👋🏾 
_This is an additional review as a result of the second opinion request._

You made a complaint about the reviewer asking you to implement tests for DOM manipulation and LocalStorage updates.
Note that the reviewers each follow a set of guidelines for these reviews. From the project requirement, it says:

> Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:
> Mock a storage object to "imitate" localStorage operations.
> Mock HTML to test if add/delete functions add or remove exactly one <li> element to/from the list in the DOM. 

What that statement means is that testing for pure functions should be straightforward and easy to write, not that you should only test for pure functions which is why the next statement suggests using mocks for the local storage operations and DOM manipulation.  So, you are to test for most pure functions and use mocks for the ones involving DOM manipulation and LocalStorage update. 

I hope this helps to fix the misunderstanding 😅 . You have already implemented most of the changes. So kindly fix your linter errors and submit them for a review.

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by benwmx** [comment]
thank you @DeliceLydia.

**Pull request merged**